gglasso/solver/single_admm_solver.py

```diff
@@ -7,154 +7,184 @@
 from scipy.sparse.csgraph import connected_components
 from scipy.linalg import block_diag
 
-
 from .ggl_helper import prox_od_1norm, phiplus, prox_rank_norm
 
 
-def ADMM_SGL(S, lambda1, Omega_0 , Theta_0 = np.array([]), X_0 = np.array([]), \
-             eps_admm = 1e-5 , rho= 1., max_iter = 1000, verbose = False, measure = False, latent = False, mu1 = None):
+def ADMM_stopping_criterion(Omega, Omega_t_1, Theta, Theta_t_1, L, X, S, tol, latent=False,
+                            mu1=None):
+    assert Omega.shape == Theta.shape == S.shape
+    assert S.shape[0] == S.shape[1]
+
+    if not latent:
+        assert np.all(L == 0)
+
+    (p, p) = S.shape
+
+    eps_rel = 1e-3
+    eps_abs = tol
+
+    psi = ((p ** 2 + p) / 2)  # number of elements of off-diagonal matrix
+    e_pri = psi * eps_abs + eps_rel * np.maximum(np.linalg.norm(Omega), np.linalg.norm(Theta))
+    e_dual = psi * eps_abs + eps_rel * np.linalg.norm(X)
+
+    r_k = np.linalg.norm(Omega - Theta)
+    s_k = np.linalg.norm(Theta - Theta_t_1)
+
+    if latent:
```

Comment:
the if clause here is redundant, if `latent=False`, then L_t is the zero array. Argument `Theta_t_1` is then also redundant, because we can use `Omega_t_1` also for the non-latent case.